### PR TITLE
runtime: removed defer processing flag and legacy codepaths.

### PR DESCRIPTION
**Pull request by KBaichoo**
Commit Message: envoy.reloadable_features.defer_processing_backedup_streams and legacy codepaths
Additional Description:
Risk Level: low, was enabled by default for ~1yr.
Testing: n/a
Docs Changes: 
Release Notes: inline
Platform Specific Features: n/a
Fixes https://github.com/envoyproxy/envoy/issues/35366

### CODE REVIEW

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by mattklein123** [comment]
Needs main merge, thanks.

/wait

File: test/integration/multiplexed_upstream_integration_test.cc [hunk 1]
```diff
@@ -361,25 +362,18 @@ TEST_P(MultiplexedUpstreamIntegrationTest, ManyLargeSimultaneousRequestWithBuffe
...
 
-TEST_P(MultiplexedUpstreamIntegrationTest, ManyLargeSimultaneousRequestWithRandomBackup) {
+TEST_P(MultiplexedUpstreamIntegrationTest, DISABLED_ManyLargeSimultaneousRequestWithRandomBackup) {
   // random-pause-filter does not support HTTP3.
   if (upstreamProtocol() == Http::CodecType::HTTP3) {
     return;
   }
 
-  if (GetParam().defer_processing_backedup_streams) {
-    // TODO(kbaichoo): fix this test to work with deferred processing by using a
```
Comment (adisuissa) [inline test/integration/multiplexed_upstream_integration_test.cc:371]: I think the comment should be tagged next to the test name - to keep the context around why the test was disabled (consider adding a tracking issue for this).

File: test/integration/multiplexed_upstream_integration_test.cc [hunk 2]
```diff
@@ -26,7 +26,8 @@ class MultiplexedUpstreamIntegrationTest : public HttpProtocolIntegrationTest {
 
   void bidirectionalStreaming(uint32_t bytes);
   void manySimultaneousRequests(uint32_t request_bytes, uint32_t max_response_bytes,
-                                uint32_t num_streams = 50);
+                                uint32_t num_streams = 50,
+                                std::chrono::milliseconds timeout = TestUtility::DefaultTimeout);
```
Comment (adisuissa) [inline test/integration/multiplexed_upstream_integration_test.cc:30]: IIUC the timeout is only used for debugging the disabled test. Let's remove it for now, as it only impacts the disabled test.

File: test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc [hunk 3]
```diff
@@ -24,26 +24,14 @@ namespace {
...
 
-std::string ipAndDeferredProcessingParamsToString(
-    const ::testing::TestParamInfo<std::tuple<Network::Address::IpVersion, bool>>& p) {
-  return fmt::format("{}_{}", TestUtility::ipVersionToString(std::get<0>(p.param)),
-                     std::get<1>(p.param) ? "WithDeferredProcessing" : "NoDeferredProcessing");
-}
-
 // TODO(kbaichoo): Remove parameterizing by deferred processing when the feature
 // is enabled by default. The parameterization is to avoid bit rot since it's
 // off by default.
```
Comment (adisuissa) [inline test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc:29]: nit: remove comment

**Review by adisuissa — COMMENTED** [review]
Minor test nits, but otherwise LGTM, thanks!

File: test/integration/multiplexed_upstream_integration_test.cc (hunk 1 above)
Comment (KBaichoo) [reply test/integration/multiplexed_upstream_integration_test.cc:371]: good idea, done.

File: test/integration/multiplexed_upstream_integration_test.cc (hunk 2 above)
Comment (KBaichoo) [reply test/integration/multiplexed_upstream_integration_test.cc:30]: done.

File: test/per_file_coverage.sh [hunk 4]
```diff
@@ -9,7 +9,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/crypto:95.5"
 "source/common/event:95.6" # Emulated edge events guards don't report LCOV
 "source/common/filesystem/posix:96.3" # FileReadToEndNotReadable fails in some env; createPath can't test all failure branches.
-"source/common/http/http2:96.0"
```
Comment (KBaichoo) [inline test/per_file_coverage.sh:12]: Coverage decrease is due to removing lines that were runtime guarded and tests were parameterized to cover them.

File: test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc [hunk 5]
```diff
@@ -1661,12 +1618,11 @@ class OverrideConfigGrpcJsonTranscoderIntegrationTest : public GrpcJsonTranscode
...
             )EOF";
     config_helper_.prependFilter(filter);
-  }
-};
-INSTANTIATE_TEST_SUITE_P(
-    IpVersionsDeferredProcessing, OverrideConfigGrpcJsonTranscoderIntegrationTest,
-    testing::Combine(testing::ValuesIn(TestEnvironment::getIpVersionsForTest()), testing::Bool()),
-    ipAndDeferredProcessingParamsToString);
+  } // namespace
+};  // namespace Envoy
```
Comment (adisuissa) [inline test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc:1622]: nit: remove `// namespace.*`

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
CI format error seems to be valid.

**Review by adisuissa — APPROVED** [review]

**Comment by RyanTheOptimist** [comment]
coverage LGTM

**Pull request merged**